internal/controller/etcdcluster_controller.go

```diff
@@ -47,17 +63,189 @@ type EtcdClusterReconciler struct {
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.19.1/pkg/reconcile
 func (r *EtcdClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+	logger := log.FromContext(ctx)
+
+	// Fetch the EtcdCluster resource
+	etcdCluster := &ecv1alpha1.EtcdCluster{}
+
+	err := r.Get(ctx, req.NamespacedName, etcdCluster)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			logger.Info("EtcdCluster resource not found. Ignoring since object may have been deleted")
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, err
+	}
+
+	if etcdCluster.Spec.Size == 0 {
+		logger.Info("EtcdCluster size is 0..Skipping next steps")
```

Comment:
nit, a warning message might a little better. It is unusual to create a EtcdCluster with 0 member.

Reply:
I am using https://github.com/go-logr/logr as the logging library for now and although it does offer verbosity levels, but as a simple library, I think it doesn't have a warning concept per se. 

We can move to a more mature logging library such as zap logger in a follow-on PR and then address this feedback comment.